spring-grpc-spring-boot-autoconfigure/src/main/java/org/springframework/grpc/autoconfigure/client/GrpcClientProperties.java

```diff
@@ -15,35 +15,415 @@
 
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Consumer;
 
 import org.springframework.boot.context.properties.ConfigurationProperties;
-import org.springframework.boot.context.properties.NestedConfigurationProperty;
-import org.springframework.grpc.client.NamedChannel;
+import org.springframework.boot.convert.DurationUnit;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.StandardEnvironment;
+import org.springframework.grpc.client.NegotiationType;
+import org.springframework.grpc.client.VirtualTargets;
+import org.springframework.util.unit.DataSize;
+
+import io.grpc.ManagedChannel;
 
 @ConfigurationProperties(prefix = "spring.grpc.client")
-public class GrpcClientProperties {
+public class GrpcClientProperties implements EnvironmentAware, VirtualTargets {
+
+	/**
+	 * The default channel configuration to use for new channels.
+	 */
+	private final ChannelConfig defaultChannel = new ChannelConfig();
 
-	@NestedConfigurationProperty
-	private final NamedChannel defaultChannel = new NamedChannel();
+	/**
+	 * Map of channels configured by name.
+	 */
+	private final Map<String, ChannelConfig> channels = new HashMap<>();
 
-	private final Map<String, NamedChannel> channels = new HashMap<>();
+	private Environment environment;
 
 	GrpcClientProperties() {
 		this.defaultChannel.setAddress("static://localhost:9090");
+		this.environment = new StandardEnvironment();
 	}
 
-	public Map<String, NamedChannel> getChannels() {
+	public ChannelConfig getDefaultChannel() {
+		return this.defaultChannel;
+	}
+
+	public Map<String, ChannelConfig> getChannels() {
 		return this.channels;
 	}
 
+	@Override
+	public void setEnvironment(Environment environment) {
+		this.environment = environment;
+	}
+
 	/**
-	 * Gets the default {@link NamedChannel} configured for the GRPC client.
-	 * @return the default {@link NamedChannel}
+	 * Gets the configured channel with the given name. If no channel is configured for
+	 * the specified name then one is created using the default channel as a template.
+	 * @param name the name of the channel
+	 * @return the configured channel if found, or a newly created channel using the
+	 * default channel as a template
 	 */
-	public NamedChannel getDefaultChannel() {
-		return this.defaultChannel;
+	public ChannelConfig getChannel(String name) {
+		if ("default".equals(name)) {
+			return this.defaultChannel;
+		}
+		ChannelConfig channel = this.channels.get(name);
+		if (channel != null) {
+			return channel;
+		}
+		channel = this.defaultChannel.copy();
+		String address = name;
+		if (!name.contains(":/") && !name.startsWith("unix:")) {
+			address = "static://" + name;
+		}
+		channel.setAddress(address);
+		return channel;
+	}
+
+	@Override
+	public String getTarget(String authority) {
```

Comment:
I found that having the getTarget next to the getChannel made the code a bit easier to follow than when it was in the NamedChannelVirtualTargets.